Index: /django/trunk/django/core/management.py
===================================================================
--- /django/trunk/django/core/management.py (revision 5102)
+++ /django/trunk/django/core/management.py (revision 5106)
@@ -169,4 +169,6 @@
         if isinstance(f, (models.ForeignKey, models.OneToOneField)):
             rel_field = f.rel.get_related_field()
+            while isinstance(rel_field, (models.ForeignKey, models.OneToOneField)):
+                rel_field = rel_field.rel.get_related_field()
             data_type = get_rel_data_type(rel_field)
         else:
