--- faker/providers/person/ro_RO/__init__.py.old	2018-02-16 04:57:48.915570971 -0800
+++ faker/providers/person/ro_RO/__init__.py	2018-02-16 04:57:51.243625893 -0800
@@ -104,6 +104,8 @@
         'Vasilică', 'Veniamin', 'Vicențiu', 'Victor', 'Vincențiu', 'Viorel', 'Visarion', 'Vlad', 'Vladimir', 'Vlaicu',
         'Voicu', 'Zamfir', 'Zeno')
 
+    first_names = first_names_male + first_names_female
+
     # sources: https://ro.wikipedia.org/wiki/Lista_celor_mai_uzuale_nume_de_familie#Rom%C3%A2nia
     last_names = (
         'Aanei', 'Ababei', 'Albu', 'Ardelean', 'Barbu', 'Cristea', 'Diaconescu', 'Diaconu', 'Dima', 'Dinu', 'Dobre',
