--- django/db/models/fields/__init__.py.old	2017-06-14 08:43:21.665812000 -0700
+++ django/db/models/fields/__init__.py	2017-12-17 14:34:03.023976702 -0800
@@ -2323,7 +2323,7 @@
         if self.has_default() and not callable(self.default):
             return self.default
         default = super(BinaryField, self).get_default()
-        if default == '':
+        if isinstance(default, six.text_type) and default == '':
             return b''
         return default
 
